pkg/analyzer/hpa.go

```diff
@@ -28,10 +28,23 @@ func (HpaAnalyzer) Analyze(a common.Analyzer) ([]common.Result, error) {
 
 		switch scaleTargetRef.Kind {
 		case "Deployment":
-			_, err := a.Client.GetClient().AppsV1().Deployments(hpa.Namespace).Get(a.Context, scaleTargetRef.Name, metav1.GetOptions{})
-			if err != nil {
-				scaleTargetRefNotFound = true
-			}
+            deployment, err := a.Client.GetClient().AppsV1().Deployments(hpa.Namespace).Get(a.Context, scaleTargetRef.Name, metav1.GetOptions{})
```

Comment:
I think we should add this on all ScaleTargetRef ( `Deployment`, `Statefulset`, etc... ). What do you think ?

Reply:
Yeah sure we can have that. Sorry I misunderstood the issue, I thought it was for just the deployment case. Also, should I add new test cases for these too?